client/src/components/list-paginator/list-paginator.js

```diff
@@ -1,25 +1,15 @@
-import ko from 'knockout';
-
 import './list-paginator.less';
 import template from 'components/list-paginator/list-paginator.html';
 import register from 'components/component-registrator';
 
 class ListPaginatorViewModel {
     constructor(params) {
-        if (params.array) {
-            this.array = params.array;
-        }
-        this.parentComponent = params.parentComponent;
         this.pageSizes = [5, 10, 15, 20];
-        this.pageSize =  ko.observable(this.pageSizes[0]);
-        this.currentPage = ko.observable(1);
-        this.displayedItems = ko.pureComputed(() => {
-            const first = (this.currentPage() - 1) * this.pageSize();
-            return this.array().slice(first, first + this.pageSize());
-        });
-        this.pageCount = ko.pureComputed(() => {
-            return Math.ceil(this.array().length / this.pageSize());
-        });
+
+        this.currentPage = params.currentPage;
+        this.pageCount = params.pageCount;
+        this.pageSize = params.pageSize;
+        this.pageSize(this.pageSizes[0]);
```

Comment:
So, here you're relying on the fact that your server has the same default page. What if someone from the team changes it?